File: Java_serial/from_file_example/get_memory_from_file.java
class get_memory_from_file {

    public get_memory_from_file() {

    }

    public static void main(String[] args) {
        
        //* Pre set values here, comment out to use command line arguments.
        String file_location = "/Users/oliverwilliams/Documents/qmul/Network_memory_project/release_builds/manufacturing_elist_comp_res_60_links_100.txt";
        String separator = "\t";
        boolean has_header = false;
        int min_order = 0;
        int max_order = 10;
        //*/

        /* Comment out to use pre set values.
        String file_location = args[1];
        String separator = args[2];
        boolean has_header = Boolean.parseBoolean(args[3]);
        int min_order = Integer.parseInt(args[4]);
        int max_order = Integer.parseInt(args[5]);
        //*/

        boolean[][] edge_series = temp_net_parser.parse_edge_file(file_location, separator, has_header);
        int order_est = memory_estimators_with_co_order_coll.get_network_effective_memory(edge_series, max_order, min_order);

        System.out.println(order_est);
    }
}

File: c++_paralell/testing/get_memory_from_file.cpp
#include <iostream>
#include <cmath>
#include <string.h> 
#include <vector>

#include "temp_net_parser_head.h"
#include "memory_estimators_with_co_order_testing_head.h"

using namespace std;
int main(int argc, char *argv[]) {
	
//	string file_location = "test_data.txt";
	
		
	string file_locaton;
	char seperator;
	int has_header;
	int min_order;
	int max_order;
	
	if(argc > 1) {
		file_locaton = argv[1];
		seperator = *argv[2];
		sscanf(argv[3],"%d",&has_header);
		sscanf(argv[4],"%d",&min_order);
		sscanf(argv[5],"%d",&max_order);
	} else {
		cout << "No arguments provided, testing case..." << endl;
		file_locaton = "/Users/oliverwilliams/Documents/qmul/Network_memory_project/release_builds/manufacturing_elist_comp_res_60_links_100.txt";
		has_header = 0;
		min_order = 0;
		max_order = 10;
	}
		
	
	vector<vector<bool> > edge_series = parse_edge_file(file_locaton, '\t', bool(has_header));
	
	cout << "got edge list from file" << endl;
	
	cout << "edges: " << edge_series.size() << endl;
	cout << "series length: " << edge_series[0].size() << endl;
	
	int memory_estimate = get_network_effective_memory(edge_series, max_order, min_order);
	
	cout << "The estimated memory is: " << memory_estimate << endl;
	

	
	return 0;
}